src/transformers/models/big_bird/modeling_flax_big_bird.py

```diff
@@ -978,25 +997,31 @@ def _bigbird_block_rand_mask(
             start = i - 2
             end = i
             if i == 1:
-                rand_attn[i - 1, :] = np.random.permutation(middle_seq[2:last])[:r]
+                rand_attn = rand_attn.at[i - 1].set(jax.random.permutation(indices_prng_key, middle_seq[2:last])[:r])
             elif i == 2:
-                rand_attn[i - 1, :] = np.random.permutation(middle_seq[3:last])[:r]
+                rand_attn = rand_attn.at[i - 1].set(jax.random.permutation(indices_prng_key, middle_seq[3:last])[:r])
             elif i == from_seq_length // from_block_size - 3:
-                rand_attn[i - 1, :] = np.random.permutation(middle_seq[:last])[:r]
+                rand_attn = rand_attn.at[i - 1].set(jax.random.permutation(indices_prng_key, middle_seq[:last])[:r])
             # Missing -3: should have been sliced till last-3
             elif i == from_seq_length // from_block_size - 2:
-                rand_attn[i - 1, :] = np.random.permutation(middle_seq[:last])[:r]
+                rand_attn = rand_attn.at[i - 1].set(jax.random.permutation(indices_prng_key, middle_seq[:last])[:r])
             # Missing -4: should have been sliced till last-4
             else:
                 if start > last:
                     start = last
-                    rand_attn[i - 1, :] = np.random.permutation(middle_seq[:start])[:r]
+                    rand_attn = rand_attn.at[i - 1].set(
+                        jax.random.permutation(indices_prng_key, middle_seq[:start])[:r]
+                    )
                 elif (end + 1) == last:
-                    rand_attn[i - 1, :] = np.random.permutation(middle_seq[:start])[:r]
+                    rand_attn = rand_attn.at[i - 1].set(
+                        jax.random.permutation(indices_prng_key, middle_seq[:start])[:r]
+                    )
                 else:
-                    rand_attn[i - 1, :] = np.random.permutation(
-                        np.concatenate((middle_seq[:start], middle_seq[end + 1 : last]))
-                    )[:r]
+                    rand_attn = rand_attn.at[i - 1].set(
+                        jax.random.permutation(
+                            indices_prng_key, jnp.concatenate((middle_seq[:start], middle_seq[end + 1 : last]))
```

Comment:
Especially when we have the concat we should definitely split over two lines for readability ✅